proxy/src/connection.rs

```diff
@@ -33,10 +35,14 @@ pub fn connect(addr: &SocketAddr,
 /// A socket that is in the process of connecting.
 pub enum Connecting {
     Plaintext {
+        addr: SocketAddr,
         connect: ConnectFuture,
         tls: Option<tls::ConditionalConnectionConfig<tls::ClientConfig>>,
     },
-    UpgradeToTls(tls::UpgradeClientToTls),
+    UpgradeToTls {
+        addr: SocketAddr,
+        upgrading: tls::UpgradeClientToTls
```

Comment:
I think this should be named `upgrade` to be consistent with `connect` in the `Plaintext` variant.